server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1709,6 +1697,31 @@ static void prepareResizeIndexSettings(
         }
     }
 
+    public static Settings.Builder copySettingsFromSource(boolean copyPrivateSettings, Settings sourceSettings, IndexScopedSettings indexScopedSettings, Settings.Builder indexSettingsBuilder) {
```

Comment:
Was trying to factor out and reused this existing code. But it required adding the `copyPrivateSettings` argument because the existing code does copy private settings, which made things more complicated? Maybe I should just leave `MetadataCreateIndexService` unchanged and add a specific method for the new use case?